java/arcs/core/data/proto/RecipeProtoDecoder.kt

```diff
@@ -17,12 +17,21 @@ import arcs.core.data.Recipe
 /** Converts a [RecipeProto] into [Recipe]. */
 fun RecipeProto.decode(particleSpecs: Map<String, ParticleSpec>): Recipe {
     val recipeHandles = mutableMapOf<String, Recipe.Handle>()
-    handlesList.forEach {
-        val oldValue = recipeHandles.put(it.name, it.decode())
+    val decodeAndUpdate = { it: HandleProto ->
```

Comment:
if you're going to provide a name for the arg for the block, something more semantically meaningful than `it` would be good.